UDV-Core/src/Extensions/Contribute/src/CreateDocument.js

```diff
@@ -0,0 +1,243 @@
+  this.initialize = function initialize()
+  {
+      this.creationContainer.innerHTML =
+          '<br/><div id = "creationTitle">Add new document</div><br/>\
+          <br/>\
+          <button id = "closeCreation">Close</button>\
+          <div class="creation">\
+          <div id = "creationWindow" ></div></div>\
+          <div id ="creationTabs">\
+          <button id = "docCreation">Send</button>\
+          <button id = "documentPositioner">Place doc</button>\
+          </div>\
+          ';
+          this.docModelToSchema();
+
+
+    var positionerContainer = document.createElement("div");
+    positionerContainer.id = "positionerContainer";
+    document.body.appendChild(positionerContainer);
+
+    positionerContainer.innerHTML =
+      '<div id="docPositionerFull">\
+      <img id="docPositionerFullImg"/></div>\
+  ';
+
+
+  var manualPositionsWindow = document.createElement("div");
+  manualPositionsWindow.id = "manualPos";
+  document.body.appendChild(manualPositionsWindow);
+
+  manualPositionsWindow.innerHTML=
+  '<div id = "inputFields" >\
+  <table>\
+  <tr><td><label id = "xPosLab" for="setPosX">XPosition </label></td>\
+  <td><input id = "setPosX"></label></td>\
+  </tr>\
+  <tr><td><label id = "yPosLab" for="setPosY">YPosition</label></td>\
+  <td><input id = "setPosY"></label></td>\
+  </tr>\
+  <tr><td><label id = "zPosLab" for="setPosZ">ZPosition</label></td>\
+  <td><input id = "setPosZ"></label></td>\
+  </tr>\
+  <tr><td><label id = "xQuatLab" for="quatX">XQuaternion</label></td>\
+  <td><input id = "quatX"></label></td>\
+  </tr>\
+  <tr><td><label id = "yQuatLab" for="quatY">YQuaternion</label></td>\
+  <td><input id = "quatY"></label></td>\
+  <tr><td><label id = "zQuatLab" for="quatZ">ZQuaternion</label></td>\
+  <td><input id = "quatZ"></label></td>\
+  </tr>\
+  <tr><td><label id = "wQuatLab" for="quatW">WQuaternion</label></td>\
+  <td><input id = "quatW"></label></td>\
+  </tr>\
+  </table>\
+  </div>\
+  <div id="docPositionerFullPanel">\
+      <button id="docPositionerSave" type=button>Save</button>\
+      <button id="moveDocument" type=button>Go to position</button>\
+      <button id = "docPositionerCancel" type=button>Cancel</button>\
+  </div>\
+  ';
+
+  }
+
+
+this.activateCreateWindow = function activateCreateWindow(active){
+  if (typeof active != 'undefined')
+  {
+      this.windowIsActive = active;
+  }
+  document.getElementById(this.contributeController.creationContainerId).style.display = active  ? "block" : "none ";
+  document.getElementById('manualPos').style.display = active  ? "block" : "none ";
+  document.getElementById('positionerContainer').style.display = active  ? "block" : "none ";
+
+
+  this.contributeController.documentController.documentResearch.activateWindow(true);
+  this.contributeController.documentController.documentBrowser.activateWindow(true);
+}
+
+this.activateDebugPosition = function activateDebugPosition(active){
```

Comment:
Is this really a debug function ? If yes it should be removed. If no it should be renamed.

Reply:
You are right, I renamed it. This function is used to set document's position manually.